payu/models/cesm_cmeps.py

```diff
@@ -190,6 +184,73 @@ def setup(self):
                 # TODO: copied this from other models. Surely we want to exit here or something
                 print('payu: error: Unable to find mod_def.ww3 file in input directory')
 
+    def _setup_checks(self):
+        # check pelayout fits within requested cpucount
+        cpucount = int(
+            self.expt.config.get('ncpus', multiprocessing.cpu_count())
+        )
+        all_realms = self.realms
+        for realm in all_realms:
+            ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
+            nthreads = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_nthreads"))
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+
+            if (pestride<nthreads):
+                raise ValueError(
+                    f"{realm}_pestride needs to be equal or greater than {realm}_nthreads"
+                )
```

Comment:
But we can have threads across multiple cores, can't we ?

Reply:
Threads can be distributed across cores of a node but each core will handle only one thread at a time. Hence there will be only one thread per core without the ability for each core to deal with multi threads at the same time.

Reply:
Yes so we need this check to ensure pestride >= nthreads so that its not trying to put more than one thread on a core?

Reply:
I dont think so, since threading is not enabled from CMEPS. And why does pestride need to be equal or greater than nthreads?

Reply:
If pestride =1 and nthreads =1 , ntasks =2 :

task0 - thread0 -> PE0
task0 - thread1 -> PE1
task1 - thread0 -> PE1
task1 -thread1 -> PE 2

-> which is problematic because theres two threads on PE1 ?

Reply:
i.e. pestride >= nthreads to enforce there is one thread per pe